programming-extensions/programming-extension-dataspaces/src/main/java/org/objectweb/proactive/extensions/dataspaces/vfs/adapter/VFSFileObjectAdapter.java

```diff
@@ -601,29 +610,19 @@ private static boolean checkIfFileWritable(DataSpacesFileObject dsfo) throws Fil
         if (!dsfo.isWritable()) {
             return false;
         }
-        File file = null;
+
         try {
-            file = convertDataSpaceToFileIfPossible(dsfo);
+            File file = convertDataSpaceToFileIfPossible(dsfo);
+
+            if (file != null) {
+                return Files.isWritable(file.toPath());
+            }
         } catch (URISyntaxException e) {
             throw new FileSystemException(e);
         } catch (DataSpacesException e) {
             throw new FileSystemException(e);
         }
-        if (file != null) {
-            boolean answer = file.canWrite();
-            if (!answer) {
-                return false;
-            }
-            // in some cases the canWrite method is not accurate, we try to create a file to check real write permission
-            File tryFile = new File(file, "testWrite.ack");
```

Comment:
Ok, I will reintroduce the original piece of code but put `tryFile.delete();` in a finally block.